pytorch_lightning/strategies/ddp.py

```diff
@@ -135,6 +134,10 @@ def num_nodes(self, num_nodes: int) -> None:
         self._num_nodes = num_nodes
         self.set_world_ranks()
 
+    @property
+    def num_processes(self):
```

Comment:
will it work for ddp2? since ddp2 inherits DDP strategy, and num_processes is num_nodes in that case set manually in L900 of `accelerator_connector`.

Reply:
accelerator_connector has self.num_processes too, but it's not equal to strategy.num_processes. They are not necessarily the same value. 

Only place accelerator_connector set strategy.num_processes is the L762-L763 before(modified below). 

The` accelerator_connector.num_processes` could be num_nodes, `parallel_devices` or `num_processes`, which equal to `trainer.num_processes` and been used in logging and profilings. 

The `Strategies.num_processes` is calculated from `strategies.parallel_devices`, which is unrelated to `accelerator_connector.num_proceses`. accelerator_connector need to set it manually base on "parallel_devices" in L762-L763 is because the `strategies.parallel_devices` is not lazy initialized

This is a pre-step for #11448, we should definitely revisit all the device related logics after #11448

Reply:
ah! you are right... got a little confused with the same name.
Thanks for clarifying :)

Reply:
Yeah, it's really confusing, and I think the devices related logic probably is wrong somewhere too. We can revisit them after the accelerator_connector rewrite